src/app/portlets/content-types/content-types-resolver.service.spec.ts

```diff
@@ -0,0 +1,118 @@
+    let dotRouterService: DotRouterService;
+
+    beforeEach(
+        async(() => {
+            TestBed.configureTestingModule({
+                providers: [
+                    ContentTypeResolver,
+                    ContentTypesInfoService,
+                    {
+                        provide: DotRouterService,
+                        useClass: DotRouterServiceMock
+                    },
+                    { provide: CrudService, useClass: CrudServiceMock },
+                    { provide: LoginService, useClass: LoginServiceMock },
+                    {
+                        provide: ActivatedRouteSnapshot,
+                        useValue: activatedRouteSnapshotMock
+                    }
+                ],
+                imports: [RouterTestingModule]
+            });
+            crudService = TestBed.get(CrudService);
+            router = TestBed.get(ActivatedRouteSnapshot);
+            contentTypeResolver = TestBed.get(ContentTypeResolver);
+            dotRouterService = TestBed.get(DotRouterService);
+        })
+    );
+
+    it('should get and return a content type', () => {
+        activatedRouteSnapshotMock.paramMap.get = () => '123';
+        spyOn(crudService, 'getDataById').and.returnValue(Observable.of({
+            fake: 'content-type',
+            object: 'right?'
+        }));
+        contentTypeResolver
+            .resolve(activatedRouteSnapshotMock)
+            .subscribe((fakeContentType: any) => {
+                expect(fakeContentType).toEqual({
+                    fake: 'content-type',
+                    object: 'right?'
+                });
+            });
+        expect(crudService.getDataById).toHaveBeenCalledWith(
+            'v1/contenttype',
+            '123'
+        );
+    });
+
+    it('should get and return null and go to home', () => {
+        activatedRouteSnapshotMock.paramMap.get = () => '123';
+        spyOn(dotRouterService, 'gotoPortlet');
+        spyOn(crudService, 'getDataById').and.returnValue(Observable.of(false));
+        contentTypeResolver
+            .resolve(activatedRouteSnapshotMock)
+            .subscribe((res: any) => {
+                expect(res).toBeNull();
+            });
+        expect(crudService.getDataById).toHaveBeenCalledWith(
+            'v1/contenttype',
+            '123'
+        );
+        expect(dotRouterService.gotoPortlet).toHaveBeenCalledWith(
+            '/content-types-angular'
+        );
+    });
+
+    it('should return a content type placeholder base on type', () => {
+        activatedRouteSnapshotMock.paramMap.get = param => {
+            return param === 'type' ? 'content' : false;
+        };
+        spyOn(dotRouterService, 'gotoPortlet');
+        spyOn(crudService, 'getDataById').and.returnValue(Observable.of(false));
+        contentTypeResolver
+            .resolve(activatedRouteSnapshotMock)
+            .subscribe((res: any) => {
+                expect(res).toEqual({
```

Comment:
are you sure that this expect always is executed?, I think you need a tick or a whenStable o something like that

Reply:
https://github.com/dotCMS/core-web/pull/447#discussion_r144920884